File: apps/frontend/angular/src/app/games/x01/Metadata.ts
import { DartDto } from './dtos/DartDto';
import { PlayerDto } from './dtos/PlayerDto';
import { GameDto } from './dtos/GameDto';

export class Metadata {
  Game!: GameDto;
  Players!: PlayerDto[];
  Darts!: Record<string, DartDto[]>;
  NextPlayer!: string;
  WinningPlayer!: string;
}

File: packages/Flyingdarts.Backend.Shared/Dtos/Metadata.cs
namespace Flyingdarts.Backend.Shared.Dtos;

public class Metadata
{
    public GameDto Game { get; set; }
    public IOrderedEnumerable<PlayerDto> Players { get; set; }
    public Dictionary<string, List<DartDto>> Darts { get; set; }
    public string NextPlayer { get; set; }
    public string WinningPlayer { get; set; }
    public Dictionary<string, object> toDictionary()
    {
        var result = new Dictionary<string, object>
        {
            { "Game", Game },
            { "Players", Players },
            { "Darts", Darts },
            { "NextPlayer", NextPlayer },
            { "WinningPlayer", WinningPlayer }
        };

        return result;
    }
}
